testcommon/instanceSmartContractCreatorTest.go

```diff
@@ -0,0 +1,61 @@
+package testcommon
+
+import (
+	"testing"
+
+	"github.com/ElrondNetwork/arwen-wasm-vm/arwen"
+	contextmock "github.com/ElrondNetwork/arwen-wasm-vm/mock/context"
+	"github.com/ElrondNetwork/elrond-go/core/vmcommon"
+)
+
+// TestCreateTemplateConfig holds the data to build a contract creation test
+type TestCreateTemplateConfig struct {
+	t             *testing.T
+	address       []byte
+	input         *vmcommon.ContractCreateInput
+	setup         func(arwen.VMHost, *contextmock.BlockchainHookStub)
+	assertResults func(*contextmock.BlockchainHookStub, *VMOutputVerifier)
+}
+
+// BuildInstanceCreatorTest starts the building process for a contract creation test
+func BuildInstanceCreatorTest(t *testing.T) *TestCreateTemplateConfig {
+	return &TestCreateTemplateConfig{
+		t:     t,
+		setup: func(arwen.VMHost, *contextmock.BlockchainHookStub) {},
+	}
+}
+
+// WithInput provides the ContractCreateInput for a TestCreateTemplateConfig
+func (callerTest *TestCreateTemplateConfig) WithInput(input *vmcommon.ContractCreateInput) *TestCreateTemplateConfig {
+	callerTest.input = input
+	return callerTest
+}
+
+// WithAddress provides the address for a TestCreateTemplateConfig
+func (callerTest *TestCreateTemplateConfig) WithAddress(address []byte) *TestCreateTemplateConfig {
+	callerTest.address = address
+	return callerTest
+}
+
+// WithSetup provides the setup function for a TestCreateTemplateConfig
+func (callerTest *TestCreateTemplateConfig) WithSetup(setup func(arwen.VMHost, *contextmock.BlockchainHookStub)) *TestCreateTemplateConfig {
+	callerTest.setup = setup
+	return callerTest
+}
+
+// AndAssertResults provides the function that will aserts the results
+func (callerTest *TestCreateTemplateConfig) AndAssertResults(assertResults func(*contextmock.BlockchainHookStub, *VMOutputVerifier)) {
+	callerTest.assertResults = assertResults
+	runTestContractCreate(callerTest)
+}
+
+func runTestContractCreate(callerTest *TestCreateTemplateConfig) {
+
+	host, stubBlockchainHook := DefaultTestArwenForDeployment(callerTest.t, 24, callerTest.address)
+	callerTest.setup(host, stubBlockchainHook)
+
+	vmOutput, err := host.RunSmartContractCreate(callerTest.input)
+
+	verify := NewVMOutputVerifier(callerTest.t, vmOutput, err)
+	callerTest.assertResults(stubBlockchainHook, verify)
```

Comment:
In a future PR, it would be good to change `TestCreateTemplateConfig` to be similar to `MockInstancesTestTemplate` with a `.runTest()` method. It is fine for now.

Reply:
done